File: 128.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> s(nums.begin(), nums.end());
        int ans = 0;
        for(int i = 0; i < nums.size(); i++){
            
            if(s.find(nums[i]-1) == s.end()){
                int l = 1;
                int x = nums[i];
                while(s.find(x+1) != s.end() ){
                    x++;
                    l++;
                }
                if(l > ans)ans = l;
            }
        }
        return ans;
    }
};

File: Python/128.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        nums = sorted(nums)
        if len(nums) == 1 or len(nums) == 0:
            return len(nums)
        last = nums[0]
        ctr = 1
        sol = 1 
        for i in range(1, len(nums)):
            if last + 1 == nums[i]:
                ctr = ctr + 1
                last = nums[i]
            elif last == nums[i]:
                continue
            else:
                sol = max(sol, ctr)
                ctr = 1
                last = nums[i]
        sol = max(ctr, sol)
        return sol
